app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt

```diff
@@ -45,9 +45,13 @@ class PresetsFragment : BaseFragment<FragmentPresetsBinding>() {
             when (val currentPosition = binding.categoryView.currentItem) {
                 categoriesAdapter.itemCount - 1 -> {
                     binding.categoryView.setCurrentItem(0, true)
+                    val currentCategory = categoriesAdapter.getCategory( 0)
+                    presetsViewModel.onCategorySelected(currentCategory)
                 }
                 else -> {
                     binding.categoryView.setCurrentItem(currentPosition + 1, true)
+                    val currentCategory = categoriesAdapter.getCategory(currentPosition+1)
+                    presetsViewModel.onCategorySelected(currentCategory)
```

Comment:
The magic is happening Inside the binding.categoryForwardButton and the binding.categoryBackButton. *sparkle sparkle* 

When these are clicked, I'm extracting the current category via a new method I created "getCategory" that uses the currentPosition Int to index the list of categories from the "items" collection found in the CategoriesPagerAdapter.

I logged - er I mean - I timber'd (ofc) and printed the currentPosition as I clicked the forward button to change categories - and I noticed the currentPosition keeps growing to a very high number - it just increments the number 4ever...so in the getCategory method, i am just going to modulo that bad boi to get a correct index.